src/ripple/net/impl/RPCCall.cpp

```diff
@@ -624,10 +624,49 @@ class RPCParser
         return parseAccountRaw1 (jvParams);
     }
 
-    // account_lines <account> <account>|"" [<ledger>]
+    // account_lines <account> <account>|"" [<ledger> [<limit>] [<marker>]]
     Json::Value parseAccountLines (Json::Value const& jvParams)
     {
-        return parseAccountRaw2 (jvParams, jss::peer);
+        std::array<char const* const, 2> accFields{{jss::account, jss::peer}};
+        auto const nParams = jvParams.size ();
+        Json::Value jvRequest (Json::objectValue);
+        for (auto i = 0; i < nParams; ++i)
+        {
+            std::string strParam = jvParams[i].asString ();
+
+            if (i==1 && strParam.empty())
+                continue;
+
+            // Parameters 0 and 1 are accounts
+            if (i < 2)
+            {
+                if (parseBase58<PublicKey> (
+                        TokenType::TOKEN_ACCOUNT_PUBLIC, strParam) ||
+                    parseBase58<AccountID> (strParam) ||
+                    parseGenericSeed (strParam))
+                {
+                    jvRequest[accFields[i]] = std::move (strParam);
+                }
+                else
+                {
+                    return rpcError (rpcACT_MALFORMED);
+                }
+            }
+            else
+            {
+                if(i==2){
+                    if (!jvParseLedger (jvRequest, strParam))
+                        return rpcError (rpcLGR_IDX_MALFORMED);
+                }else{
+                    if(strParam.size()==64){
+                        jvRequest[jss::marker]=std::move (strParam);
+                    }else{
+                        jvRequest[jss::limit]=jvParams[i].asInt ();
+                    }
+                }
+            }
+        }
+        return jvRequest;
```

Comment:
I think the code is clearer if we used a switch statement in the loop. Here's my take for what that would look like: https://github.com/seelabs/rippled/commit/0972409dee177f21f165b3e95212a6c229b1ad3b